File: python/1040.py
n1, n2, n3, n4 = input().split()
n1 = float(n1)
n2 = float(n2)
n3 = float(n3)
n4 = float(n4)

avg = ((n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1)) / (2 + 3 + 4 + 1)

print(f'Media: {avg:.1f}')

if avg >= 7:
    print('Aluno aprovado.')
elif avg >= 5 and avg < 7:
    print('Aluno em exame.')
    nExam = float(input())
    print(f'Nota do exame: {nExam:.1f}')
    newAvg = (avg + nExam) / 2
    if newAvg >= 5:
        print('Aluno aprovado.')
    else:
        print('Aluno reprovado.')
    print(f'Media final: {newAvg:.1f}')
else:
    print('Aluno reprovado.')

File: lua/1040.lua
local entry = io.read()
local t = {}
for value in string.gmatch(entry, '[^%s]+') do
    table.insert(t, tonumber(value))
    
end
local avg = (t[1] * 2 + t[2] * 3 + t[3] * 4 + t[4])/10
print("Media: "..string.format('%.1f', avg))
if avg >= 7 then
    print('Aluno aprovado.')
elseif avg >= 5  then
    print("Aluno em exame.")
    local grade = io.read('n')
    avg = (avg + grade)/2
    print('Nota do exame: '..string.format('%.1f', grade))
    if avg >= 5 then
        print('Aluno aprovado.')
    else
        print('Aluno reprovado.')
    end
    print('Media final: '..string.format('%.1f', avg))
else
    print("Aluno reprovado.")
end